python/chemiscope/jupyter.py

```diff
@@ -58,6 +57,19 @@ def save(self, path):
         file.write(json.dumps(data).encode("utf8"))
         file.close()
 
+    def __repr__(self, max_length=32):
+        # string representation of the chemiscope widget, truncating too large outputs
+        class_name = self.__class__.__name__
+        attributes = vars(self)
+        string_repr = ""
+
+        for key, value in attributes.items():
+            value_repr = repr(value)
```

Comment:
I don't see how this helps with memory consumption: we are computing the full repr unconditionally here.

Reply:
the problem is that in some cases (e.g. when loading a previously-saved notebook) the whole string representation would be shown in the browser, killing the tab. this makes sure that the string representation of the widget remains sane.

Reply:
ok this was indeed generating pretty nonensical representation, missing all the important bits. I have cleaned it up, now it does IMO the right thing and avoids generating a huge string with the whole dataset in it.